src/interpreter/vm/VM.cpp

```diff
@@ -34,14 +36,14 @@ int VM::run() {
                 push(DurianObject(*reinterpret_cast<int32_t*>(p_headerStr), p_headerStr + sizeof(int32_t)));
                 break;
             case Opcode::ICONST:
-                push(*reinterpret_cast<int64_t*>(m_code+m_pc));
+                push(DurianObject(*reinterpret_cast<int64_t*>(m_code+m_pc), DurianType::Integer));
                 m_pc += sizeof(int64_t);
                 break;
             case Opcode::ICONST_0:
-                push((int64_t)0);
+                push(DurianObject((int64_t)0, DurianType::Integer));
```

Comment:
Not sure if I’m understanding the ocnstructor correctly but isn’t the default type integer already?

Reply:
We had to change it, because when we have user defined functions, we'll also need to pass in an Integer, so we need some way to tell these two apart, hence passing in the type as well for the integer constructor.

Reply:
I think what he means is that `type` is an optional parameter, with integer as the default. Personally I think we should avoid making it default/optional, since integers and functions are probably roughly as common as each other (and thus it doesn't make sense to make integer the default).